index.js

```diff
@@ -434,12 +435,13 @@
       _extractors[tuple[0]] = tuple[1];
       return _extractors;
     }, {});
+    t.arity = arity;  // number of type parameters
     t.extractors = Z.map (B (toArray), t._extractors);
     t.format = format || function(outer, inner) {
       return Z.reduce (function(s, tuple) {
         return s +
                outer (' ') +
-               when (tuple[2].keys.length > 0 && tuple[2].type !== RECORD)
+               when (tuple[2].arity > 0)
```

Comment:
The problem with using `type.keys.length` is that record types can have many keys but should *not* be considered parameterized types. By using `type.arity` we need not remember to exclude record types.